builder/dockerfile/dispatchers.go

```diff
@@ -152,13 +153,18 @@ func (d *dispatchRequest) getImageMount(imageRefOrID string) (*imageMount, error
 //
 func initializeStage(d dispatchRequest, cmd *instructions.Stage) error {
 	d.builder.imageProber.Reset()
-	if err := system.ValidatePlatform(&cmd.Platform); err != nil {
-		return err
+
+	if len(strings.TrimSpace(cmd.Platform.OS)) != 0 {
+		if _, err := platforms.Parse(cmd.Platform.OS); err != nil {
```

Comment:
Isn't `cmd.Platform.OS` already parsed somewhere before this?

Reply:
I don't think so, here the `initializeState()` parse the `platform` option embedded in the `FROM [--platform=platform]` stage, so it should be the first time to catch the value of the `--platform`, so we need a parser here to guarantee the sanity of the `platform`.